src/Magnum/Math/Test/RandomTest.cpp

```diff
@@ -0,0 +1,112 @@
+#include <Corrade/TestSuite/Tester.h>
+#include <Corrade/Utility/DebugStl.h>
+
+#include "Magnum/Math/Random.h"
+
+namespace Magnum
+{
+namespace Math
+{
+
+namespace Test
+{
+namespace
+{
+
+struct RandomTest : Corrade::TestSuite::Tester
+{
+    explicit RandomTest();
+
+    void unsignedScalar();
+    void signedScalar();
+    void unitVector2();
+    void unitVector3();
+    void pointInACircle();
+    void pointInASphere();
+    void randomRotation();
+};
+
+typedef Vector<2, Float> Vector2;
+typedef Vector<3, Float> Vector3;
+typedef Math::Constants<Float> Constants;
+
+RandomTest::RandomTest()
+{
+    addTests({&RandomTest::unsignedScalar,
+              &RandomTest::signedScalar,
+              &RandomTest::unitVector2,
+              &RandomTest::unitVector3,
+              &RandomTest::pointInACircle,
+              &RandomTest::pointInASphere,
+              &RandomTest::randomRotation});
+}
+
+// void RandomTest::construct() {
+//     constexpr Matrix4x4 a = {Vector4(3.0f,  5.0f, 8.0f, -3.0f),
+//                              Vector4(4.5f,  4.0f, 7.0f,  2.0f),
+//                              Vector4(1.0f,  2.0f, 3.0f, -1.0f),
+//                              Vector4(7.9f, -1.0f, 8.0f, -1.5f)};
+//     CORRADE_COMPARE(a, Matrix4x4(Vector4(3.0f,  5.0f, 8.0f, -3.0f),
+//                                  Vector4(4.5f,  4.0f, 7.0f,  2.0f),
+//                                  Vector4(1.0f,  2.0f, 3.0f, -1.0f),
+//                                  Vector4(7.9f, -1.0f, 8.0f, -1.5f)));
+
+//     CORRADE_VERIFY((std::is_nothrow_constructible<Matrix4x4, Vector4, Vector4, Vector4, Vector4>::value));
+// }
+void RandomTest::unsignedScalar()
+{
+    for (std::size_t i = 0; i < 10; i++)
```

Comment:
Instead of a loop in every test you can replace `addTests()` with [addRepeatedTests()](https://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Tester.html#a06c40ddf62db02c1575d019b630d7f08) and set the repeat count there. I think repeating every test 100 times won't hurt, 10 is too little :)

Reply:
Top !